cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -808,8 +808,13 @@ def _add_cfn_volume(self, id: str, shared_ebs: SharedEbs):
 
     def _add_wait_condition(self):
         wait_condition_handle = cfn.CfnWaitConditionHandle(self, id="HeadNodeWaitConditionHandle" + self.timestamp)
+        head_node_bootstrap_timeout = get_attr(self.config, "dev_settings.timeouts.head_node_bootstrap_timeout")
         wait_condition = cfn.CfnWaitCondition(
-            self, id="HeadNodeWaitCondition" + self.timestamp, count=1, handle=wait_condition_handle.ref, timeout="1800"
+            self,
+            id="HeadNodeWaitCondition" + self.timestamp,
+            count=1,
+            handle=wait_condition_handle.ref,
+            timeout=str(head_node_bootstrap_timeout) if head_node_bootstrap_timeout else "1800",
```

Comment:
if `head_node_bootstrap_timeout` already sets a default let's not have a default in multiple places.

Reply:
If DevSettings is not set, head_node_bootstrap_timeout from DevSettings will be None, default bootstrap timeout is needed here. Encapsulated it in a constant